### DISCUSSION TITLE
PDOException: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'ecommerceframework_productindex' doesn't exist

### DISCUSSION THREAD

**Discussion by ZaneCEO** [pimcore/pimcore#14258] [Q&A / Support] [answered]
Hi! My app runs a custom import procedure:

````php
....
(new \Pimcore\Model\DataObject\Product())
  ->setName('product1')
  ....
  ...save()
````

On `save()`, I get this error:

`PDOException: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'ecommerceframework_productindex' doesn't exist`

My own Product class extends `Pimcore\Bundle\EcommerceFrameworkBundle\Model\AbstractProduct`:

````php
....
class Product extends AbstractProduct
{
   ...
    public function getPriceSystemName(): ?string
    {
        return 'default';
    }
   ....
}
````

The table actually doesn't exists. All I have, related to the Pimcore Ecommerce Framework is:

````
ecommerceframework_cart
ecommerceframework_cartcheckoutdata
ecommerceframework_cartitem
ecommerceframework_pricing_rule
ecommerceframework_vouchertoolkit_reservations
ecommerceframework_vouchertoolkit_statistics
ecommerceframework_vouchertoolkit_tokens
````

I already tried to `symfony console doctrine:migrations:migrate`, but I get

`[OK] Already at the latest version`

I tried to uninstall and reinstall the Pimcore Ecommerce Framework via admin, but it didn't solve the issue.

What am I missing?

**Comment by twin-elements** [comment, marked as answer]
Try to run `php bin/console ecommerce:indexservice:bootstrap --create-or-update-index-structure`

**Reply by archanagoyal07** [reply]
Hello,

I am facing same issue. To resolve issue when I am running command, it is throwing error as below

 sudo docker compose exec php bin/console ecommerce:indexservice:bootstrap --create-or-update-index-structure

In ParallelExecutorFactory.php line 434:

  Warning: Undefined array key "PWD"

ecommerce:indexservice:bootstrap [-p|--processes [PROCESSES]] [-m|--main-process] [--child] [--batch-size BATCH-SIZE] [--segment-size SEGMENT-SIZE] [--timeout [TIMEOUT]] [--create-or-update-index-structure] [--update-index] [--object-list-class OBJECT-LIST-CLASS] [--list-condition [LIST-CONDITION]] [--tenant TENANT] [--] [<item>]

**Comment by ZaneCEO** [comment]
@twin-elements You'e THE MAN! It worked and my problem is now fixed!

Thank you so much!
